pandas/core/indexes/datetimes.py

```diff
@@ -562,6 +562,13 @@ def unique(self, level=None):
         result = super(DatetimeIndex, naive).unique(level=level)
         return self._shallow_copy(result.values)
 
+    def _is_compatible_with_other(self, other):
+        is_compat = super(DatetimeIndex, self)._is_compatible_with_other(other)
+        if not is_compat:
+            is_compat = (hasattr(other, 'dtype') 
```

Comment:
why are you checking .base here? again why is the super method not sufficient here?

Reply:
This is so that tz aware dtypes can be considered compatible, since the union between mismatched tz dtypes will default to union the arrays as UTC.

    pd.date_range('19910905', periods=10, tz='US/Eastern').dtype.base == pd.date_range('19910905', periods=10, tz='US/Central').dtype.base